build-tools-internal/src/main/java/org/elasticsearch/gradle/internal/dependencies/patches/PatcherInfo.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+package org.elasticsearch.gradle.internal.dependencies.patches;
+
+import org.objectweb.asm.ClassVisitor;
+import org.objectweb.asm.ClassWriter;
+
+import java.util.Arrays;
+import java.util.HexFormat;
+import java.util.function.Function;
+
+public record PatcherInfo(String jarEntryName, byte[] classSha256, Function<ClassWriter, ClassVisitor> visitorFactory) {
+
+    /**
+     * Creates a patcher info entry, linking a jar entry path name to a patcher factory (a factory to create an ASM visitor)
+     * @param jarEntryName the jar entry path, as a string
+     * @param visitorFactory the factory to create an ASM visitor from a ASM writer
+     */
+    public static PatcherInfo classPatcher(String jarEntryName, Function<ClassWriter, ClassVisitor> visitorFactory) {
+        return new PatcherInfo(jarEntryName, null, visitorFactory);
```

Comment:
Why not define `classSha256` everywhere? If we upgrade the HDFS lib the same concerns about checking the new patchers work also apply.

Reply:
I did not want to introduce too many changes in this PR.
I plan to do that in a separate PR as an enhancement.

Reply:
Perhaps do this first? It's less of a behavioural change, just a preparatory step for this one.

Reply:
Even better. Let me separate this and the extraction `patchJar` into `Utils` into a preliminary PR

Reply:
PR is here: https://github.com/elastic/elasticsearch/pull/126449